src/approaches/pg3_approach.py

```diff
@@ -46,14 +47,39 @@ def __init__(self, initial_predicates: Set[Predicate],
     def get_name(cls) -> str:
         return "pg3"
 
-    def _predict(self, state: State, atoms: Set[GroundAtom],
-                 goal: Set[GroundAtom]) -> _GroundNSRT:
-        del state  # unused
+    def predict(self, atoms: Set[GroundAtom],
+                goal: Set[GroundAtom]) -> _GroundNSRT:
+        """Predicts next GroundNSRT to be deployed based on the PG3 generated
+        policy."""
         ground_nsrt = utils.query_ldl(self._current_ldl, atoms, goal)
         if ground_nsrt is None:
             raise ApproachFailure("PG3 policy was not applicable!")
         return ground_nsrt
 
+    def solve(self, task: Task, timeout: int) -> Callable[[State], Action]:
```

Comment:
I think we actually want to be overriding `_solve`, not `solve` -- check the parent classes